nav2_dwb_controller/dwb_critics/src/oscillation.cpp

```diff
@@ -172,19 +172,19 @@ void OscillationCritic::debrief(const nav_2d_msgs::msg::Twist2D & cmd_vel)
     }
   }
 }
-
 bool OscillationCritic::resetAvailable()
 {
   if (oscillation_reset_dist_ >= 0.0) {
-    double x_diff = pose_.x - prev_stationary_pose_.x;
-    double y_diff = pose_.y - prev_stationary_pose_.y;
+    double x_diff = pose_.position.x - prev_stationary_pose_.position.x;
+    double y_diff = pose_.position.y - prev_stationary_pose_.position.y;
     double sq_dist = x_diff * x_diff + y_diff * y_diff;
     if (sq_dist > oscillation_reset_dist_sq_) {
       return true;
     }
   }
   if (oscillation_reset_angle_ >= 0.0) {
-    double th_diff = pose_.theta - prev_stationary_pose_.theta;
+    double th_diff = tf2::getYaw(pose_.orientation) -
```

Comment:
We should be using the angles min distance - I know this isn't a bug you introduced but this is a bug we should fix

Reply:
I have added this solution:

```
  if (oscillation_reset_angle_ >= 0.0) {
    tf2::Quaternion tf2_q1, tf2_q2;
    tf2::fromMsg(pose_.orientation, tf2_q1);
    tf2::fromMsg(prev_stationary_pose_.orientation, tf2_q2);

    tf2::Quaternion delta_q = tf2_q2 * tf2_q1.inverse();
    double th_diff = tf2::getYaw(delta_q);

    if (fabs(th_diff) > oscillation_reset_angle_) {
      return true;
    }
  }
```

Let me know if you like other solution.